plugin/eqdetect/source/dextool/plugin/eqdetect/runner.d

```diff
@@ -0,0 +1,105 @@
+/**
+Copyright: Copyright (c) 2017, Joakim Brännström. All rights reserved.
+License: MPL-2
+Author: Joakim Brännström (joakim.brannstrom@gmx.com)
+
+This Source Code Form is subject to the terms of the Mozilla Public License,
+v.2.0. If a copy of the MPL was not distributed with this file, You can obtain
+one at http://mozilla.org/MPL/2.0/.
+
+This file contains an example plugin that demonstrate how to:
+ - interact with the clang AST
+ - print the AST nodes when debugging is activated by the user via the command line
+ - find all free functions and generate a dummy implementation.
+
+It is purely intended as an example to get started developing **your** plugin.
+*/
+
+module dextool.plugin.runner;
+
+import std.stdio;
+import logger = std.experimental.logger;
+
+import dextool.type : ExitStatusType, FileName, AbsolutePath;
+
+ExitStatusType runPlugin(string[] args) {
+    RawConfiguration pargs;
+    pargs.parse(args);
+
+    if (pargs.shortPluginHelp) {
+        writeln("eqdetect");
+        writeln("Find for-loops in given file");
+        return ExitStatusType.Ok;
+    } else if (pargs.help) {
+        pargs.printHelp;
+        return ExitStatusType.Ok;
+    } else if (pargs.file.length == 0) {
+        writeln("Missing file --in");
+        return ExitStatusType.Errors;
+    }
+
+    import std.file;
+    import std.conv : to;
+    import std.algorithm : canFind;
+
+    File file = File(to!string(pargs.file), "r");
+    int i = 1;
+    while (!file.eof()) {
+        string s = file.readln();
+        if(canFind(s, "for") && !canFind(s, "//")){
```

Comment:
creative solution. But I would suggest to use the AST to familiarize with the visitor and clang.